enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,498 @@
+
+Since this is intended to be a replacement for configure-ovs, it must
+work in a cross-platform manner. The baremetal feature is implemented in the
+baremetal-operator, which means non-baremetal platforms cannot use it.
+
+However, this is not a replacement for the baremetal feature (and any other
+platform-specific deploy-time network configuration tools). This is because
+minimal networking is required in order to pull Ignition, and since this
+configuration will be provided through Ignition it will still be necessary
+to have platform-specific methods of injecting network config before Ignition
+runs. However, the platform-specific part can be much simpler because it
+won't need to handle every possible network configuration, just enough to
+reach the Ignition endpoint. In some use cases, platform-specific configuration
+won't even be needed.
+
+### Workflow Description
+
+When the user is populating install-config they will provide the necessary
+configuration in the networking section (TODO: Is this the right place?).
+It will look something like this:
+
+networking:
+  networkType: OVNKubernetes
+  machineNetwork:
+  - cidr: 192.0.2.0/24
+  [...]
+  hostConfig:
+  - name: master-0
+    networkConfig:
+      interfaces:
+      - name: enp2s0
+        type: ethernet
+        state: up
+        ipv4:
+          enabled: false
+        ipv6:
+          enabled: false
+      - name: br-ex
+        type: ovs-bridge
+        state: up
+        copy-mac-from: enp2s0
+        ipv4:
+          enabled: false
+          dhcp: false
+        ipv6:
+          enabled: false
+          dhcp: false
+        bridge:
+          port:
+          - name: enp2s0
+          - name: br-ex
+      - name: br-ex
+        type: ovs-interface
+        state: up
+        ipv4:
+          enabled: true
+          dhcp: true
+        ipv6:
+          enabled: false
+          dhcp: false
+
+The name field will need to be something that uniquely identifies a given host.
+I'm unsure if name is necessarily the best way to do that, but it is one valid
+option. I reused the networkConfig name from the baremetal feature for
+consistency. Perhaps something different would be preferable to avoid confusion
+though?
+
+Note that although the configuration can be provided on a per-host basis, it is
+not mandatory to do so. YAML anchors can be used to duplicate a single
+configuration across multiple nodes. For example:
+
+networking:
+  networkType: OVNKubernetes
+  machineNetwork:
+  - cidr: 192.0.2.0/24
+  [...]
+  hostConfig:
+  - name: master-0
+    networkConfig: &BOND
+      interfaces:
+      - name: br-ex
+        ...etc...
+  - name: master-1
+    networkConfig: *BOND
+  - name: master-2
+    networkConfig: *BOND
+
+This reduces duplication in install-config and the chance of errors due to
+unintended differences in node config.
+
+When this mechanism is in use, a flag will be set to indicate to configure-ovs
```

Comment:
Oops, did I not reply to this before?

The answer is yes. configure-ovs will still run in the default case, but if you've provided custom br-ex config it will prevent configure-ovs from running.